zjit/src/backend/lir.rs

```diff
@@ -1889,9 +1889,8 @@ impl Assembler {
         out
     }
 
-    pub fn add_into(&mut self, left: Opnd, right: Opnd) -> Opnd {
+    pub fn add_into(&mut self, left: Opnd, right: Opnd) {
```

Comment:
I don't love that this breaks SSA if we use a vreg operand as an argument. Can we require it only take memory/phyreg? (here and below)

Reply:
I changed them to only take `Opnd::Reg` for now https://github.com/ruby/ruby/pull/14177/commits/a43e62e9eaa1304abc61f67cc6ac7826e8bad2ed since it's the only expected usage of them.

Btw if you want LIR to be SSA, you need to fix `cpop_into`, `lea_into`, `load_into`, `mov`, and `store` as well.

Reply:
Almost. You can make it happen with just ^. We just haven't come up with anything that needs the guarantee yet.